packages/react/src/enhancers/withServerPlaceholder.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { ComponentType } from 'react';
+import { ServerPlaceholder } from '../components/Placeholder';
+import { ComponentRendering } from '@sitecore-content-sdk/core/layout';
+import { ComponentMap } from '../components/sharedTypes';
+import { Page } from '@sitecore-content-sdk/core/client';
+
+export type ComponentProps = {
+  rendering: ComponentRendering;
+  placeholders: Record<string, React.ReactNode>;
+  componentMap?: ComponentMap;
+};
+
+export type WrapperProps = {
+  page: Page;
+  componentMap?: ComponentMap;
+  rendering: ComponentRendering;
+};
+
+export const withServerPlaceholder = <T extends ComponentProps, W extends T & WrapperProps>(
+  Component: ComponentType<T>,
+  placeholders: string[]
+) => {
+  return (props: W) => {
+    const phProps: Record<string, unknown> = {};
```

Comment:
consider wrapping this with useMemo since it gets re-created on every render.

Reply:
useMemo can't be used with server components unfortunately:
https://github.com/vercel/next.js/discussions/68071#discussioncomment-10134807